test/suites/smoke-test-common-solo/test-invulnerables-priority.ts

```diff
@@ -47,28 +48,88 @@ describeSuite({
                 if (collators.length <= invulnerables.length) {
                     // Less collators than invulnerables - all collators must be invulnerables
                     for (const collator of collators) {
-                        expect(invulnerables.includes(collator), `Collator should be in invulnerable list: ${collator}`)
-                            .to.be.true;
+                        expect(
+                            invulnerables.includes(collator),
+                            `[#${currentBlockNumber}] Collator should be in invulnerable list: ${collator}`
+                        ).to.be.true;
                     }
                 } else {
                     // More collators than invulnerables: all invulnerables must be collators
                     for (const invulnerable of invulnerables) {
-                        expect(
-                            collators.includes(invulnerable),
-                            `Invulnerable should be in collators list: ${invulnerable}`
-                        ).to.be.true;
+                        if (!collators.includes(invulnerable)) {
+                            // Sometimes if the chain is de-registered, the invulnerable is not in the collators list
+                            // we check if it was in the list after the full_rotation
+                            await checkIfInvulnerableWasAssignedAfterFullRotation(api, invulnerable);
+                        }
                     }
 
                     // Remaining collators must be from staking
                     const collatorsNotInvulnerables = collators.filter((collator) => !invulnerables.includes(collator));
                     for (const collator of collatorsNotInvulnerables) {
                         expect(
                             eligibleCandidates.includes(collator),
-                            `Collator should be a staking candidate: ${collator}`
+                            `[#${currentBlockNumber}] Collator should be a staking candidate: ${collator}`
                         ).to.be.true;
                     }
                 }
             },
         });
     },
 });
+
+export const checkIfInvulnerableWasAssignedAfterFullRotation = async (
+    api: ApiPromise,
+    invulnerable: string
+): Promise<void> => {
+    const epochDuration = api.consts.babe.epochDuration.toNumber();
+    const fullRotationSessions = 24; // for both Dancelight and Starlight
+
+    const epochStart = (await api.query.babe.epochStart())[0].toNumber();
+
+    for (let i = 1; i <= fullRotationSessions; i++) {
+        const sessionFirstBlock = epochStart - i * epochDuration;
```

Comment:
change this please. you can always fetch the beginning block of the session by querying 

`(await api.query.babe.epochStart())[0].toNumber();`

like you do at the beginning. therefore you go to the first block of the session, and if its not the full-rotation, then you should go back one block and query the same storage item

Reply:
Fixed:
https://github.com/moondance-labs/tanssi/pull/1348/commits/2c08c4e29b899d799be7320b49e5720176785c1c